File: AdminPanel1/AdminPanel1.Web/Modules/Product/Product/ProductForm.cs
using Serenity.ComponentModel;
using System;

namespace AdminPanel1.Product.Forms;

[FormScript("Product.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public int CatId { get; set; }
    public string Photo { get; set; }
    public string Type { get; set; }
    public string SupplierName { get; set; }
    public DateTime EntryDate { get; set; }
    public string ReviewUrl { get; set; }
    public int Quantity { get; set; }
}

File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/Product/ProductForm.ts
import { StringEditor, DecimalEditor, LookupEditor, ImageUploadEditor, DateEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ProductForm {
    Name: StringEditor;
    Description: StringEditor;
    Price: DecimalEditor;
    CatId: LookupEditor;
    Photo: ImageUploadEditor;
    Type: StringEditor;
    SupplierName: StringEditor;
    EntryDate: DateEditor;
    ReviewUrl: StringEditor;
    Quantity: IntegerEditor;
}

export class ProductForm extends PrefixedContext {
    static readonly formKey = 'Product.Product';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProductForm.init)  {
            ProductForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = LookupEditor;
            var w3 = ImageUploadEditor;
            var w4 = DateEditor;
            var w5 = IntegerEditor;

            initFormType(ProductForm, [
                'Name', w0,
                'Description', w0,
                'Price', w1,
                'CatId', w2,
                'Photo', w3,
                'Type', w0,
                'SupplierName', w0,
                'EntryDate', w4,
                'ReviewUrl', w0,
                'Quantity', w5
            ]);
        }
    }
}